cypress/plugins/testDataProcessor.js

```diff
@@ -0,0 +1,390 @@
+const fs = require('fs');
+const jsonMerger = require('json-merger');
+const CONSTANTS = require('../support/constants/constants');
+const REGEXFORMATS = require('../support/constants/regexformats');
+let envVariables;
+const path = require('path');
+const _ = require('lodash');
+const logger = require('../support/Logger')('testDataProcessor.js');
```

Comment:
Proper comments and function descriptions to be added for all functions
Each if/else block to have proper comments

Reply:
addressed the comment